modules/coredns/collect.go

```diff
@@ -89,13 +81,34 @@ func (cd *CoreDNS) collect() (map[string]int64, error) {
 	return stm.ToMap(mx), nil
 }
 
-// Summary
+func (cd *CoreDNS) updateVersionDependentMetrics(raw prometheus.Metrics) {
+	if cd.version == nil {
+		if version := cd.Version(raw); version != nil {
+			cd.version = version
+			if cd.version.LTE(version169) {
+				cd.metricNames.panicCountTotal = metricPanicCountTotal169orOlder
+				cd.metricNames.requestCountTotal = metricRequestCountTotal169orOlder
+				cd.metricNames.requestTypeCountTotal = metricRequestTypeCountTotal169orOlder
+				cd.metricNames.responseRcodeCountTotal = metricResponseRcodeCountTotal169orOlder
+			} else {
+				cd.metricNames.panicCountTotal = metricPanicCountTotal170orNewer
+				cd.metricNames.requestCountTotal = metricRequestCountTotal170orNewer
+				cd.metricNames.requestTypeCountTotal = metricRequestTypeCountTotal170orNewer
+				cd.metricNames.responseRcodeCountTotal = metricResponseRcodeCountTotal170orNewer
+			}
+		}
+	}
+}
 
 func (cd CoreDNS) Version(raw prometheus.Metrics) *semver.Version {
```

Comment:
I suggest making this method private or even make it a function ("parseVersion" or something).

Reply:
done!